core/chains/evm/logpoller/log_poller.go

```diff
@@ -375,15 +377,16 @@ func (lp *logPoller) pollAndSaveLogs(ctx context.Context, currentBlockNumber int
 	// E.g. 1<-2<-3(currentBlockNumber)<-4<-5<-6<-7(latestBlockNumber), finality is 2. So 3,4 can be batched.
 	// Although 5 is finalized, we still need to save it to the db for reorg detection if 6 is a reorg.
 	// start = currentBlockNumber = 3, end = latestBlockNumber - finality - 1 = 7-2-1 = 4 (inclusive range).
-	if (latestBlockNumber - currentBlockNumber) >= (lp.finalityDepth + 1) {
-		lp.lggr.Infow("Backfilling logs", "start", currentBlockNumber, "end", latestBlockNumber-lp.finalityDepth-1)
-		currentBlockNumber = lp.backfill(ctx, currentBlockNumber, latestBlockNumber-lp.finalityDepth-1)
+	lastSafeBackfillBlock := latestBlockNumber - lp.finalityDepth - 1
+	if lastSafeBackfillBlock >= currentBlockNumber {
+		lp.lggr.Infow("Backfilling logs", "start", currentBlockNumber, "end", lastSafeBackfillBlock)
+		currentBlockNumber = lp.backfill(ctx, currentBlockNumber, lastSafeBackfillBlock)
 	}
 
 	for currentBlockNumber <= latestBlockNumber {
 		// Same reorg detection on unfinalized blocks.
-		currentBlock, err2 := lp.getCurrentBlockMaybeHandleReorg(ctx, currentBlockNumber)
-		if err2 != nil {
+		currentBlock, err1 = lp.getCurrentBlockMaybeHandleReorg(ctx, currentBlockNumber)
+		if err1 != nil {
```

Comment:
This was a bug as err1 is used below instead of err2 when logging the error. Since currentBlock is already declared before we can use `=` instead of `:=` and actually assign to err1 to match the logging.

Reply:
agree on the logging, but I think it should remain named err2 just to align with the general structure of nesting err names I have